### DISCUSSION TITLE
Ability To Reduce Cookie Size by Excluding Metadata

### DISCUSSION THREAD

**Discussion by J43fura** [supabase/supabase#29782] [Feature Requests]
### Ability To Reduce Cookie Size by Excluding Metadata

**Summary:**
I would like to request the ability of a reduction in the cookie size for Supabase Auth by excluding metadata from being stored in the cookie. By default, Supabase Auth currently stores metadata in the session cookie, leading to larger cookie sizes that can negatively impact performance & user experience, especially in environments where cookie size limits are a concern.

**Motivation:**
1. **Compliance with Cookie Size Limits:** Many web browsers and reverse proxies impose strict limits on cookie sizes (often around 8KB). Exceeding this limit can lead to unexpected behaviors, including rejection of cookies (python-websockets, Nuxt, Cloudflare...) causing **431 Request Header Fields Too Large** issues.
   
2. **Better Developer Control:** Allowing developers to disable the storing of metadata in the cookie if necessary will give developers better control over the size and contents of the authentication cookies.

**Proposed Solution:**
- **Optional disable:** Provide a configuration option for developers to disable metadata storage in cookies if needed.

**Additional Considerations:**
- For applications that require metadata, developers could access it securely from the server side or via API calls instead of storing it in cookies.

**Related issues:**
- https://github.com/supabase/supabase-js/issues/963#issuecomment-2050423845
- https://github.com/supabase/ssr/issues/56
- https://github.com/supabase/auth/issues/1754
- https://github.com/supabase/supabase-js/issues/963

---

Thank you for considering this request! Reducing the default cookie size and providing flexibility to developers will enhance the performance and scalability of applications using Supabase Auth.

**Comment by MichaelJCole** [comment]
Hello, this is also being flagged in the OpenResty Web Application Firewall rules as a hack.  

It's a problem...

**Comment by sam-jg** [comment]
Looks like this will be released soon, including for SSR.

https://github.com/supabase/supabase-js/pull/1545
https://github.com/supabase/ssr/pull/126
